src/data/gods/god.ts

```diff
@@ -1,20 +1,30 @@
-import { BoonType } from "./abilityTypes";
+import { StandardBoonType, InfusionBoonType, INFUSION } from "./abilityTypes";
 import { BoonElement } from "./elements";
 import { BoonRarity } from "./rarities";
 
 export type BoonValues = Partial<
   Record<BoonRarity, { [level: number]: string | number }>
 >;
 
-export type Boon = {
+export type Boon = StandardBoon | InfusionBoon;
+
+type StandardBoon = {
   name: string;
-  type: BoonType;
+  type: StandardBoonType;
   element?: BoonElement;
   info: (value: string) => string;
   values: BoonValues;
   prerequisites?: Array<Array<Boon>>;
 };
 
+export const isInfusion =
+  (boon: Boon): boon is InfusionBoon => boon.type === INFUSION
+
+export type InfusionBoon = Omit<StandardBoon, "type" | "requiredElements"> & {
```

Comment:
I think you meant to omit `prerequisites` here, not `requiredElements` since those don't exist on the Boon to begin with, I'm surprised this isn't throwing a type error actually